File: 1346. Check If N and Its Double Exist/main.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        checked_set = set()
        for element in arr:
            if element * 2 in checked_set or element / 2 in checked_set:
                return True
            checked_set.add(element)

        return False

File: 1346. Check If N and Its Double Exist/main.cpp
#include <stddef.h>

#include <vector>

class Solution {
  public:
    bool checkIfExist(vector<int> &arr) {
        size_t arr_size = arr.size();
        for (size_t i = 0; i < arr_size; i++) {
            for (size_t j = i + 1; j < arr_size; j++) {
                if ((arr[i] == arr[j] * 2) || (arr[j] == arr[i] * 2)) {
                    return true;
                }
            }
        }

        return false;
    }
};